File: Data Structures & Algorithms/find-minimum-in-rotated-sorted-array/submission-2.cpp
class Solution {
public:
    int findMin(vector<int> &nums) {
        int l = 0;
        int r = nums.size() - 1;
        while (l < r) {
            int mid = l+(r-l)/2;
            int mn = nums[mid];
            if (mn < nums[r]) r = mid;
            else l = mid + 1;
        }
        return nums[l];
        
    }
};

File: Data Structures & Algorithms/search-2d-matrix/submission-2.py
class Solution:
    def searchMatrix(self, matrix: List[List[int]], target: int) -> bool:
        l = 0
        cols = len(matrix[0])
        r = len(matrix) * cols - 1
        while (l <= r):
            mid = l + (r-l)//2
            mn = matrix[mid//cols][mid%cols]
            if (mn == target): return True
            elif (mn < target): l = mid + 1
            else: r = mid - 1
        return False